controllers/openstackmachine_controller.go

```diff
@@ -359,32 +359,34 @@ func (r *OpenStackMachineReconciler) reconcileNormal(ctx context.Context, scope
 		return ctrl.Result{RequeueAfter: waitForInstanceBecomeActiveToReconcile}, nil
 	}
 
-	if openStackCluster.Spec.APIServerLoadBalancer.Enabled {
-		err = r.reconcileLoadBalancerMember(scope, openStackCluster, machine, openStackMachine, instanceNS, clusterName)
-		if err != nil {
-			handleUpdateMachineError(scope.Logger, openStackMachine, errors.Errorf("LoadBalancerMember cannot be reconciled: %v", err))
-			return ctrl.Result{}, nil
-		}
-	} else if util.IsControlPlaneMachine(machine) && !openStackCluster.Spec.DisableAPIServerFloatingIP {
-		floatingIPAddress := openStackCluster.Spec.ControlPlaneEndpoint.Host
-		if openStackCluster.Spec.APIServerFloatingIP != "" {
-			floatingIPAddress = openStackCluster.Spec.APIServerFloatingIP
-		}
-		fp, err := networkingService.GetOrCreateFloatingIP(openStackMachine, openStackCluster, clusterName, floatingIPAddress)
-		if err != nil {
-			handleUpdateMachineError(scope.Logger, openStackMachine, errors.Errorf("Floating IP cannot be got or created: %v", err))
-			return ctrl.Result{}, nil
-		}
-		port, err := computeService.GetManagementPort(openStackCluster, instanceStatus)
-		if err != nil {
-			err = errors.Errorf("getting management port for control plane machine %s: %v", machine.Name, err)
-			handleUpdateMachineError(scope.Logger, openStackMachine, err)
-			return ctrl.Result{}, nil
-		}
-		err = networkingService.AssociateFloatingIP(openStackMachine, fp, port.ID)
-		if err != nil {
-			handleUpdateMachineError(scope.Logger, openStackMachine, errors.Errorf("Floating IP cannot be associated: %v", err))
-			return ctrl.Result{}, nil
+	if util.IsControlPlaneMachine(machine) {
+		if openStackCluster.Spec.APIServerLoadBalancer.Enabled {
+			err = r.reconcileLoadBalancerMember(scope, openStackCluster, machine, openStackMachine, instanceNS, clusterName)
+			if err != nil {
+				handleUpdateMachineError(scope.Logger, openStackMachine, errors.Errorf("LoadBalancerMember cannot be reconciled: %v", err))
+				return ctrl.Result{}, nil
+			}
```

Comment:
Does this really makes any difference? `r.reconcileLoadBalancerMember` is a no-op as it returns early `nil` if it's no control plane machine.

Reply:
Okay, I overlooked that we're calling the private method `reconcileLoadBalancerMember` and not `ReconcileLoadBalancerMember` the only difference is, that `loadbalancer.NewService()` could return an error.

https://github.com/kubernetes-sigs/cluster-api-provider-openstack/blob/2eb85ef3b49a959cb56c56ec5399f3e5bbef8627/controllers/openstackmachine_controller.go#L500-L503

Reply:
It does not make a huge difference right now, but I would like to set the condition `APIServerIngressReadyCondition` right after calling `r.reconcileLoadBalancerMember`, and that only makes sense if the machine actually was a control plane machine.

See #1288 for context of this change